src/Workspaces/Remote/ServiceHub/Services/RemoteLanguageClient/RemoteLanguageClient.cs

```diff
@@ -21,7 +21,10 @@
 
 namespace Microsoft.CodeAnalysis.Remote
 {
-    internal class LanguageServer : ServiceBase
+    /// <summary>
+    /// Remote side of the VisualStudioOOPConnectionLanguageClient.  Used to implement the ILanguageClient in the OOP server.
+    /// </summary>
+    internal class RemoteLanguageClient : ServiceBase
```

Comment:
I think this should be perhaps RemoteLanguageServer?  The interface implementation of ILanguageClient returns a connection to this, but this isn't actually itself _implementing_ ILanguageClient.  It's an LSP server that happens to be activated by the VS client (thourgh ILanguageClient).  Basically the OOP equivalent of the InProcLanguageServer (so these should at least be consistent).

I see ILanguageClient as an extension of the VS LSP client - anything on the other side of ActivateAsync is part of the LSP server.  For example there's a Rust ILanguageClient that just returns a connection to the third party Rust language server.  And this class is just the OOP C# Language server

Reply:
that seems reasonable :)